privacyidea/api/validate.py

```diff
@@ -323,6 +323,11 @@ def check():
                         "realm": user.realm})
 
     if serial:
+        if user:
+            # check if the given token belongs to the user
+            token_list = get_tokens(user=user, serial=serial)
```

Comment:
I think this should work! Maybe we could even pass ``count=True`` to only issue a ``SELECT COUNT(*)`` query, which could help performance . But I'm not so sure it would be noticeable :)

Reply:
fixed in 23816b3a

Reply:
I talke the liberty of resolving this conversation! Looks good.